FreeRTOS/FreeRTOS-Plus-TCP
# Improve Default Macros

Improve Default Macros

Description
-----------

- Cleaned, organized and added documentation to both IPTraceMacroDefaults.h & FreeRTOSIPConfigDefaults.h.
- Removed a few macros that were duplicated in FreeRTOSIPConfigDefaults.h.
- Delete FreeRTOS_errno_TCP.h as all but pdFREERTOS_ERRNO_EAFNOSUPPORT have been included in projdefs.h for quite some time. A PR has been created in [FreeRTOS-Kernel](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/642/files) to add the last remaining macro. Until then, it resides in FreeRTOSIPConfigDefaults.h.
- Created a new file, FreeRTOSIPDeprecatedDefinitions.h, that contains the checks for usage of deprecated macros. This is similar to FreeRTOS-Kernel's deprecated_definitions.h.
- Added a few missing deprecated macro checks to FreeRTOSIPDeprecatedDefinitions.h.
- Include IPTraceMacroDefaults.h in FreeRTOSIPConfigDefaults.h below the tracing enabling macros and removed all inclusions of IPTraceMacroDefaults.h elsewhere. This ensures the tracing enabling macros are always defined before including the tracing implementation macros.
- Removed some unused deprecated macros (i.e. ipconfigRAND32) from the tests' FreeRTOSIPConfig.h files.

Test Steps
-----------

N/A

Checklist:
----------



Related Issue
-----------

N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
